Fix relative links in README (#617)
epiforecasts/EpiNow2: README.Rmd

<details> <summary> Package website </summary>

The package has two websites: one for
[the stable release version on CRAN](https://epiforecasts.io/EpiNow2/), and
another for [the version in development](https://epiforecasts.io/EpiNow2/dev).
These two provide various resources for learning about the package, including
the function reference, details about each model (model definition), workflows
for each model (usage), and case studies or literature of applications of
the package. However, the development website may contain experimental features
and information not yet available in the stable release.

</details>

<details> <summary> End-to-end workflows </summary>

The workflow vignette (see `vignette("estimate_infections_workflow")`)
provides guidance on the end-to-end process of estimating reproduction
numbers and performing short-term forecasts for a disease spreading in a

</details>

<details> <summary> Model definitions </summary>

> On the websites, we provide the mathematical definition of each model.

Reviewer: ```suggestion
In different vignettes we provide the mathematical definition of each model.
```

For example, the model definition vignette for `estimate_infections()` can be

> found in `vignettes("estimate_infections")`).

Reviewer: ```suggestion
found in `vignettes("estimate_infections")`.
```

<details> <summary> End-to-end workflows </summary>

> The [workflow vignette](./articles/estimate_infections_workflow.html) provides guidance on the end-to-end process of estimating reproduction numbers and performing short-term forecasts for a disease spreading in a given setting.

Author: I couldn't figure out why the original links are not working.

Reviewer: and this fixed it?

Author: No, it didn't. I don't think it is easy to make the links in the README relative. I didn't find anything useful in the pkgdown documentation. I think we may have to use the absolute links of either the dev or stable version.

Reviewer: Could we use `vignette` statements instead? This means we can't have link text so will need some minor re-writing but I'm looking at the `pkgdown` documentation as an example:

The end of this line in the README
https://github.com/r-lib/pkgdown/blob/676cf1dce39e4dc5aa1bcb671b1e14e992589a0a/README.Rmd#L23
gets rendered to a simple `vignette("pkgdown")` in the README but a link [`vignette("pkgdown")`](https://pkgdown.r-lib.org/articles/pkgdown.html) in pkgdown.

In the line highlighted above this could be something like:
```markdown
The workflow vignette `vignette("estimate_infections_workflow")` provides guidance on the end-to-end process of estimating reproduction numbers and performing short-term forecasts for a disease spreading in a given setting.
```
or similar.

Author: That should work. See changes here 84175eb6b1ea1050dfe1892a74bf2376c2e2a966. I've also reworded the part about the website to refer explicitly to the stable and dev sites here c743acc90a438808e59a339d15705ee28a72deca.
